modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/subscription/AndesSubscription.java

```diff
@@ -425,20 +426,27 @@ public UUID getValue(AndesSubscription sub, QueryOptions queryOptions) {
         }
     };
 
-    public static final Attribute<AndesSubscription, String> ROUTER_NAME = new SimpleAttribute<AndesSubscription,
-            String>("routerName") {
+    public static final Attribute<AndesSubscription, String> ROUTER_NAME =
+            new SimpleNullableAttribute<AndesSubscription, String>("routerName") {
         @Override
         public String getValue(AndesSubscription sub, QueryOptions queryOptions) {
-            return sub.storageQueue.getMessageRouter().getName();
+            String routerName = null;
+            if (sub.storageQueue.getMessageRouter() != null) {
```

Comment:
Yes. This is getting null in subscription sync. It calls SubscriptionRegistry#removeSubscription. This removes AndesSubscription from cqengine IndexedCollection. Remove method propagate to cqengine query method and trying to access the null object. Before doing subscription sync, we do message router sync, queue sync, and binding sync and these are setting the object reference to null.